src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,670 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2016 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+/*
+ * The driver is derived from the KRT2-driver. This version has implemented two methods yet:
+ * setting the active frequency
+ * setting the passive frequency
+ * It can be added:
+ * setting/reading dual scan on/off
+ * reading frequency change on the radio
+ * setting/reading squelsh settings
+ *
+ */
+
+#include "Device/Driver/AR62xx.hpp"
+#include "Device/Driver.hpp"
+#include "Device/Port/Port.hpp"
+#include "NMEA/Info.hpp"
+#include "RadioFrequency.hpp"
+#include "Thread/Cond.hxx"
+#include "Thread/Mutex.hxx"
+#include "Util/CharUtil.hxx"
+#include "Util/StaticFifoBuffer.hxx"
+#include "LogFile.hpp"
+
+#include <stdint.h>
+#include <stdio.h>
+#include <chrono>
+#include <cstdlib>
+
+//Einige Konstanten und Ausdruecke, das AR62xx hat ein Binaerprotokoll
+constexpr uint8_t HEADER_ID = 0xA5;
+#define PROTID  0x14;
+#define QUERY   BIT(7)
+#define DUAL    BIT(8)
+#define SQUELCH BIT(7)
+#define MAX_CMD_LEN 128
+#define ACTIVE_STATION  1
+#define PASSIVE_STATION 0
+#define RoundFreq(a) ((int)((a)*1000.0+0.5)/1000.0)
+#define Freq2Idx(a) (int)(((a)-118.0) * 3040/(137.00-118.0)+0.5)
+#define BIT(n) (1 << (n))
+#define NAME_SIZE 30
+
+typedef union {
+  uint16_t intVal16;
+  uint8_t  intVal8[2];
+} IntConvertStruct;
+
+typedef struct {
```

Comment:
No `typedef struct`!
For API documentation, use doxygen-style comments, not `//`
All variable names violate XCSoar's coding style.

Reply:
removed